internal/search/query/transformer.go

```diff
@@ -563,3 +563,67 @@ func FuzzifyRegexPatterns(nodes []Node) []Node {
 		return Parameter{Field: field, Value: value, Negated: negated, Annotation: annotation}
 	})
 }
+
+// mapRevFilters removes rev: filters from []Node and attaches their value as @rev to the repo: filters.
+// All rev: filters are assumed to be on the top-level, i.e. mapRevFilter will not traverse the tree.
+// To be compatible with the output of the parser with and without DNF we handle both, []Node with
+// an explicit top-level And operator and []Node with an implicit And.
+func mapRevFilters(nodes []Node) ([]Node, error) {
+	if len(nodes) == 0 {
+		return nodes, nil
+	}
+	// top-level And
+	if op, isOperator := nodes[0].(Operator); len(nodes) == 1 && isOperator && op.Kind == And {
+		nodes, err := mapRevFiltersFlat(op.Operands)
+		if err != nil {
+			return nil, err
+		}
+		return newOperator(nodes, And), nil
+	}
+	return mapRevFiltersFlat(nodes)
+}
+
+func mapRevFiltersFlat(nodes []Node) ([]Node, error) {
+	var revs []string
+	for _, n := range nodes {
+		if param, ok := n.(Parameter); !ok || param.Field != FieldRev {
+			continue
+		} else {
+			revs = append(revs, param.Value)
+		}
+	}
+	if len(revs) == 0 {
+		return nodes, nil
+	}
+	if len(revs) > 1 {
+		return nil, fmt.Errorf("invalid syntax. You have specified multiple revisions (%s) and "+
+			" I don't know how to interpret this. Remove all but one rev: keywords"+
```

Comment:
```suggestion
		return nil, fmt.Errorf("invalid syntax. You have specified multiple revisions (%s) and"+
			" I don't know how to interpret this. Remove all but one rev: keywords"+
```

Or something to avoid the double space?